storage-operator/pkg/controller/volume/volume_controller.go

```diff
@@ -74,9 +246,237 @@ var _ reconcile.Reconciler = &ReconcileVolume{}
+		ctx, nodeName, volume.Spec.RawBlockDevice.DevicePath,
+	)
+	return *resource.NewQuantity(size, resource.BinarySI), err
+}
+
+// Get the PersistentVolume associated to the given volume.
+//
+// Return `nil` if no such volume exists.
+func (self *ReconcileVolume) getPersistentVolume(
+	ctx context.Context, name string,
+) (*corev1.PersistentVolume, error) {
+	pv := &corev1.PersistentVolume{}
+	key := types.NamespacedName{Namespace: "", Name: name}
+
+	if err := self.client.Get(ctx, key, pv); err != nil {
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return pv, nil
+}
+
+// Remove the volume-protection on the PV (if not already present).
+func (self *ReconcileVolume) removePvFinalizer(
+	ctx context.Context, pv *corev1.PersistentVolume,
+) error {
+	finalizers := pv.GetFinalizers()
+	pv.SetFinalizers(SliceRemoveValue(finalizers, VOLUME_PROTECTION))
+	return self.client.Update(ctx, pv)
+}
+
+// Create a PV associated to the volume.
+//
+// Set the ownership and finalizers to tie the PV to the volume.
+func (self *ReconcileVolume) createPersistentVolume(
+	ctx context.Context,
+	pv *corev1.PersistentVolume,
+	volume *storagev1alpha1.Volume,
+) error {
+	reqLogger := log.WithValues("Request.Name", volume.Name)
+
+	// Set Volume instance as the owner and controller.
+	err := controllerutil.SetControllerReference(volume, pv, self.scheme)
+	if err != nil {
+		return err
+	}
+	// Set volume-protection finalizer on the volume.
+	if err := self.addVolumeFinalizer(ctx, volume); err != nil {
+		reqLogger.Error(err, "cannot set volume-protection: requeue")
+		return err
+	}
+	// Create the PV!
+	if err := self.client.Create(ctx, pv); err != nil {
+		reqLogger.Error(
+			err, "cannot create PersistentVolume: requeue",
+			"PersistentVolume.Name", volume.Name,
+		)
+		return err
+	}
+	return nil
+}
+
+type stateSetter func(
+	context.Context, *storagev1alpha1.Volume, string,
+) (reconcile.Result, error)
+
+// Poll a Salt state job.
+func (self *ReconcileVolume) pollSaltJob(
+	ctx context.Context,
+	jobName string,
+	volume *storagev1alpha1.Volume,
+	setState stateSetter,
+) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Name", volume.Name)
+	nodeName := string(volume.Spec.NodeName)
+
+	jid := volume.Status.Job
+	if result, err := self.salt.PollJob(ctx, jid, nodeName); err != nil {
+		reqLogger.Error(err, fmt.Sprintf("failed to poll %s status", jobName))
+		// This one is not retryable.
+		if failure, ok := err.(*salt.AsyncJobFailed); ok {
+			self.recorder.Eventf(
+				volume, corev1.EventTypeWarning, "SaltCall",
+				"Salt call to %s failed", jobName,
+			)
+			return self.setFailedVolumeStatus(
+				ctx, volume, storagev1alpha1.CreationError,
+				"%s failed with %s", jobName, failure.Error(),
+			)
+		}
+		// Job salt not found, let's retry.
+		return setState(ctx, volume, "")
+	} else {
+		if result == nil {
+			reqLogger.Info(fmt.Sprintf("%s still in progress", jobName))
```

Comment:
Same as above, make clear what this random string that'll go in "%s" is about.

Reply:
Done